galasa-ui/src/tests/components/UserRoleSelection.test.tsx

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+import React from 'react';
+import { render, screen, fireEvent, waitFor, act } from '@testing-library/react';
+import UserRoleSection from '../../components/users/UserRoleSection';
+import { UserData } from '@/generated/galasaapi';
+
+// --- Mocks for Carbon Components and ErrorPage --- //
+jest.mock('@carbon/react', () => ({
+  ButtonSet: ({ children }: { children: React.ReactNode }) => <div>{children}</div>,
+  Button: (props: any) => <button {...props}>{props.children}</button>,
+  Dropdown: (props: any) => (
```

Comment:
Are we not able to run tests without mocking the carbon components out? Seems like we're not testing what actually appears since we're replacing the carbon components with vanilla HTML